File: Demo/Application/Features/PayPal - Checkout/BraintreeDemoPayPalCheckoutViewController.m
#import "BraintreeDemoPayPalCheckoutViewController.h"
@import BraintreePayPal;

@implementation BraintreeDemoPayPalCheckoutViewController

- (UIView *)createPaymentButton {
    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
    [button setTitle:NSLocalizedString(@"PayPal Checkout", nil) forState:UIControlStateNormal];
    [button setTitleColor:[UIColor blueColor] forState:UIControlStateNormal];
    [button setTitleColor:[UIColor colorWithRed:50.0/255 green:50.0/255 blue:255.0/255 alpha:1.0] forState:UIControlStateHighlighted];
    [button setTitleColor:[UIColor lightGrayColor] forState:UIControlStateDisabled];
    [button addTarget:self action:@selector(tappedPayPalCheckout:) forControlEvents:UIControlEventTouchUpInside];
    return button;
}

- (void)tappedPayPalCheckout:(UIButton *)sender {
    self.progressBlock(@"Tapped PayPal - checkout using BTPayPalDriver");

    [sender setTitle:NSLocalizedString(@"Processing...", nil) forState:UIControlStateDisabled];
    [sender setEnabled:NO];

    BTPayPalDriver *driver = [[BTPayPalDriver alloc] initWithAPIClient:self.apiClient];
    BTPayPalCheckoutRequest *request = [[BTPayPalCheckoutRequest alloc] initWithAmount:@"4.30"];
    request.activeWindow = self.view.window;
    [driver tokenizePayPalAccountWithPayPalRequest:request completion:^(BTPayPalAccountNonce * _Nullable payPalAccount, NSError * _Nullable error) {
        [sender setEnabled:YES];

        if (error) {
            self.progressBlock(error.localizedDescription);
        } else if (payPalAccount) {
            self.completionBlock(payPalAccount);
        } else {
            self.progressBlock(@"Canceled");
        }
    }];
}

@end

File: Demo/Application/Features/PayPal - Checkout/BraintreeDemoPayPalCheckoutViewController.swift
import UIKit
import BraintreePayPal

class BraintreeDemoPayPalCheckoutViewController: BraintreeDemoPaymentButtonBaseViewController {
    
    override func createPaymentButton() -> UIView! {
        lazy var payPalCheckoutButton: UIButton = {
            let payPalCheckoutButton = UIButton(type: .system)
            payPalCheckoutButton.setTitle("PayPal Checkout", for: .normal)
            payPalCheckoutButton.setTitleColor(.blue, for: .normal)
            payPalCheckoutButton.setTitleColor(.lightGray, for: .highlighted)
            payPalCheckoutButton.setTitleColor(.lightGray, for: .disabled)
            payPalCheckoutButton.addTarget(self, action: #selector(tappedPayPalCheckout), for: .touchUpInside)
            return payPalCheckoutButton
        }()
        
        return payPalCheckoutButton
    }
    
    @objc func tappedPayPalCheckout(_ sender: UIButton) {
        progressBlock("Tapped PayPal - Checkout using BTPayPalClient")
        sender.setTitle("Processing...", for: .disabled)
        sender.isEnabled = false
        
        let client = BTPayPalClient(apiClient: apiClient)
        let request = BTPayPalCheckoutRequest(amount: "4.30")
        
        client.tokenize(request) { nonce, error in
            sender.isEnabled = true
            
            guard let nonce = nonce else {
                self.progressBlock(error?.localizedDescription)
                return
            }
            self.completionBlock(nonce)
        }
    }
}
